src/tiled/objecttypeseditor.h

```diff
@@ -23,7 +23,13 @@
 #include "properties.h"
 
 #include <QDialog>
-
+#include <QColorDialog>
+#include <QCloseEvent>
+#include <QFileDialog>
+#include <QInputDialog>
+#include <QMessageBox>
+#include <QPainter>
```

Comment:
Seems to me like you need none of these includes, only the `QStyledItemDelegate` include should be added.